pyriemann/transferlearning.py

```diff
@@ -0,0 +1,147 @@
+from .classification import MDM
+import numpy as np
+from joblib import Parallel, delayed
+
+from .utils.mean import mean_covariance
+from .utils.geodesic import geodesic
+
+class MDWM (MDM):
+    """Classification by Minimum Distance to Weighted Mean.
+
+    Classification by nearest centroid. For each of the given classes, a
+    centroid is estimated, according to the chosen metric, as a weighted mean
+    of point (i.e. covariance matrices) from the source domain, combined with
+    the class centroid of the target domain.
+    For classification, a given new point is attibuted to the class whose
+    centroid is the nearest according to the chosen metric.
+
+    Parameters
+    ----------
+    metric : string | dict (default: 'riemann')
+        The type of metric used for centroid and distance estimation.
+        see `mean_covariance` for the list of supported metric.
+        the metric could be a dict with two keys, `mean` and `distance` in
+        order to pass different metric for the centroid estimation and the
+        distance estimation. Typical usecase is to pass 'logeuclid' metric for
+        the mean in order to boost the computional speed and 'riemann' for the
+        distance in order to keep the good sensitivity for the classification.
+    L : float, (default: 0)
+        Transfer coefficient in [0,1]. This parameter controls the trade-off
+        between source and target data.
+    n_jobs : int, (default: 1)
+        The number of jobs to use for the computation. This works by computing
+        each of the class centroid in parallel.
+        If -1 all CPUs are used. If 1 is given, no parallel computing code is
+        used at all, which is useful for debugging. For n_jobs below -1,
+        (n_cpus + 1 + n_jobs) are used. Thus for n_jobs = -2, all CPUs but one
+        are used.
+
+    Attributes
+    ----------
+    covmeans_ : list
+        the class centroids.
+    classes_ : list
+        list of classes.
+
+    See Also
+    --------
+    MDM
+    Kmeans
+    FgMDM
+    KNearestNeighbor
+
+    References
+    ----------
+    [1] E. Kalunga, S. Chevallier and Q. Barthélemy, "Transfer learning for
+    SSVEP-based BCI using Riemannian similarities between users", in 26th
+    European Signal Processing Conference (EUSIPCO), pp. 1685-1689. IEEE, 2018.
+
+    [2] S. Khazem, S. Chevallier, Q. Barthélemy, K. Haroun and C. Noûs,
+    "Minimizing Subject-dependent Calibration for BCI with Riemannian Transfer
+    Learning", in 10th International IEEE/EMBS Conference on Neural
+    Engineering (NER), pp. 523-526. IEEE, 2021.
+    """
+
+    def __init__(self, metric='riemann', L=0, n_jobs=1):
```

Comment:
Parameters checking must be in `fit` instead of `init`, in the case parameters have been modified between by `set_parameters`.

Reply:
I don't know what is the common practice here. But intuitively, I would add another validation in method `set_parameters` too. This way the user is not allowed to proceed from the moment they set up wrong parameters.